nncf/initialization.py

```diff
@@ -210,6 +224,8 @@ def set_bn_momentum(module, momentum_value):
             args_kwargs_tuple = data_loader.get_inputs(loaded_item)
             self._infer_batch(args_kwargs_tuple, device)
 
+        self.model.eval()
```

Comment:
Not sure we should always set the model to eval mode. It can be not obvious for the users of this init functionality.

I'd add a context manager or decorator for secure safe switching modes.

The idea is always to keep the original state of the model with activating the chosen mode inside a function's body. 
```
def _run_model_inference
    # doesn't matter what is a current training mode of the model, it would be returned to the original state outside of the context manager
    with training_mode_switcher(model, is_training=True) as model:
         # work with the model in a training mode
         model.infer()
    # outside of context manager model is in the original sate
    model.infer()
```

That would be helpful in multiple places in NNCF.

Reply:
@vanyalzr are you going to address this?

Reply:
Sure, I agree with the comment, will address